src/librustc_middle/ty/context.rs

```diff
@@ -926,12 +928,11 @@ pub struct GlobalCtxt<'tcx> {
     pub(crate) untracked_crate: &'tcx hir::Crate<'tcx>,
     pub(crate) definitions: &'tcx Definitions,
 
-    /// A map from `DefPathHash` -> `DefId`. Includes `DefId`s from the local crate
-    /// as well as all upstream crates. Only populated in incremental mode.
-    pub def_path_hash_to_def_id: Option<FxHashMap<DefPathHash, DefId>>,
-
     pub queries: query::Queries<'tcx>,
 
+    pub local_def_path_hash_to_def_id: Option<FxHashMap<DefPathHash, DefId>>,
+    def_path_hash_to_def_id_cache: Lock<FxHashMap<DefPathHash, Option<DefId>>>,
```

Comment:
Could you add a comment telling that this map includes local def ids from the `local_def_path_hash_to_def_id` above? Othewise it looks like this map is for foreign def ids only.

Reply:
This map includes both local and foreign `DefPathHashes` - we use it to cache all lookups of a `DefPathHash`. I'll add a comment explaining tihs.